### DISCUSSION TITLE
Replication from Raven 3 to Raven 5

### DISCUSSION THREAD

**Discussion by RecoX** [ravendb/ravendb#16789] [Q&A] [answered]
Hello dear Raven friends, 

I have RavenDB 3 in a VM using Windows Server 2016 and I am currently migrating to Raven 5

I have the following error trying to replicate the data from a RavenDB 3 to RavenDB 5 hosted in the cloud with a cluster.

In RavenDB 5 Dashboard in the section `Manage Server->Certificates` I generated a client certificate `replication35ok` Thumbprint: `1234567809D610D1123455750F33C12BEXXXXX`

The certificate has ADMIN permissions for all the DBs (they are empty and created with the same names as in Raven 3)

In Raven 3 my Raven.Server.exe.config looks like this:
```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
    <appSettings>
        <add key="Raven/Port" value="*"/>
        <add key="Raven/DataDir/Legacy" value="~\Database\System"/>
        <add key="Raven/DataDir" value="~\Databases\System"/>
        <add key="Raven/AnonymousAccess" value="Admin"/>
        <add key="Raven/Licensing/AllowAdminAnonymousAccessForCommercialUse" value="false" />
        <add key="Raven/AccessControlAllowOrigin" value="*" />
        <add key="Raven/UseSsl" value="true" />
        <add key="Raven/Replication/CertificatePath" value="C:\replication35ok.pfx" />
    </appSettings>
    <runtime>
        <loadFromRemoteSources enabled="true"/>
        <gcServer enabled="true"/>
        <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
            <probing privatePath="Analyzers;Plugins"/>
        </assemblyBinding>
    </runtime>
</configuration>
```

But I can't connect

```
Failed to contact replication destination: https://a.something.mybrand.ravendb.cloud/databases/InStoreApps InStoreApps
Exception 
System.Net.WebException: Error:  ---> System.Net.WebException: The remote server returned an error: (403) Forbidden.

   at System.Net.HttpWebRequest.GetResponse()

   at Raven.Abstractions.Connection.HttpRavenRequest.SendRequestToServer(Action`1 action) in C:\Builds\RavenDB-Stable-3.5\Raven.Abstractions\Connection\HttpRavenRequest.cs:line 219

   --- End of inner exception stack trace ---

   at Raven.Abstractions.Connection.HttpRavenRequest.SendRequestToServer(Action`1 action) in C:\Builds\RavenDB-Stable-3.5\Raven.Abstractions\Connection\HttpRavenRequest.cs:line 249

   at Raven.Abstractions.Connection.HttpRavenRequest.ExecuteRequest[T]() in C:\Builds\RavenDB-Stable-3.5\Raven.Abstractions\Connection\HttpRavenRequest.cs:line 160

   at Raven.Bundles.Replication.Tasks.ReplicationTask.GetLastReplicatedEtagFrom(ReplicationStrategy destination) in C:\Builds\RavenDB-Stable-3.5\Raven.Database\Bundles\Replication\Tasks\ReplicationTask.cs:line 1749
Logged

4 minutes ago (07/03/23, 1:07pm)
Level

Warn
Logger

Raven.Bundles.Replication.Tasks.ReplicationTask
```

I did the same with another environment where I got Raven 3 installed (development) and it works fine, but there are two differences.

The development server uses Windows Server 2019 and checking the TLS supported versions by .NET shows the following:

[Enum]::GetValues([System.Net.SecurityProtocolType]) | Where-Object { $_ -match 'Tls' }
```
Tls
Tls11
Tls12
Tls13
```

The staging environment (the one that does not work and throws 403 error) uses Windows Server 2016 and it throws the follow TLS list

[Enum]::GetValues([System.Net.SecurityProtocolType]) | Where-Object { $_ -match 'Tls' }
```
Tls
Tls11
Tls12
```

Is it possible that is not working for the Tls13 and the replication will not work from Raven 3 to 5  using Windows Server 2016? 

What can I do to solve this issue? 

Thanks

**Comment by ayende** [comment]
What version of 3.x are you running?

**Comment by RecoX** [comment]
I see, thanks for your reply.

There is another difference in the environments

The Raven 3 with the 403 problems has:

    Server Build #35262
    Client Build #35262


The Raven 3 working with replication just fine has the following:

    Server Build #35312
    Client Build #35312

So probably is this, right?

**Reply by ayende** [reply, marked as answer]
Yes, please use the latest 3.5 version

**Comment by RecoX** [comment]
Resolved the issue by upgrading to the latest RavenDB 3.5 version.
